File: examportal/src/main/java/com/exam/service/QuizService.java
package com.exam.service;

import java.util.Set;

import com.exam.model.Quiz;

public interface QuizService {
	
	public Quiz addQuiz(Quiz quiz);
	
	public Quiz updateQuiz(Quiz quiz);
	
	public Set<Quiz> getQuizzes();
	
	public Quiz getQuiz(Long quizId);
	
	public void deleteQuizById(Long quizId);
	
}

File: examportal_frontend/exam-front/src/app/services/quiz.service.ts
import { Injectable } from '@angular/core';
import { HttpClient} from '@angular/common/http';
import baseUrl from './helper';
import { Quiz } from '../model/quiz';
@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http: HttpClient) { }


  addQuiz(quiz:Quiz) {
    return this.http.post(`${baseUrl}/quiz/`,quiz);
  }

  updateQuiz(quiz:Quiz) {
    return this.http.put(`${baseUrl}/quiz/`,quiz);
  }

  getQuizzes() {
    return this.http.get(`${baseUrl}/quiz/`);
  }

  getQuizById(quizId:number) {
    return this.http.get(`${baseUrl}/quiz/${quizId}`);
  }

  deleteQuizById(quizId:number) {
    return this.http.delete(`${baseUrl}/quiz/${quizId}`);
  }

}
